### PR TITLE
CI/CD Documentation

### PR DESCRIPTION
**Pull request by cshannon2**
Initial Commit for CI/CD Documentation

### CODE REVIEW

**Comment by cshannon2** [comment]
I am not sure how thorough we want to be with CI/CD given that our current workflow is utilizing JHU Assets. This also brings up the question of whether this CI/CD pipeline is currently ready for us to share in a public repository vs having this be placed in a JHU specific repository.

**Comment by cshannon2** [comment]
> May want to think about incorporating these documents into CI/CD documentation:
> 
> https://github.com/eclipse-pass/main/blob/main/docs/dev/release-steps-with-automations.md https://github.com/eclipse-pass/main/blob/main/docs/dev/release.md

I think this initial document is currently written as an overview section / landing page, that will transition the reader into reading more about the release process. We can structure the documentation to lead into the release and automation documents that you are referencing.

**Comment by rpoet-jh** [comment]
Consider adding a page for the following workflows that are in the JHU repo:

- Staging Deployment: Workflow that runs on a schedule and checks if any of the PASS docker images have changed from what is deployed in PASS staging.  If a docker image changes, it gets deployed. This handles automatic deployment to stage when PRs are merged.  The workflow can also be run manually.

- Deployment Tests: Workflow that can be manually run to execute the pass-acceptance tests against a live PASS environment. Test data is cleaned up within 24 hours from PASS. Downstream deposits can be turned on/off for deployment tests. For JHU, we currently do deposits into stage jscholarship for deployment tests and delete the deposit from stage jscholarship within 1 hour.   For deployment tests that run against PROD, we don't deposit into jscholarship.

Adding these would just be a description of how each workflow works since the community would not have access to the workflow files in the JHU repo. It would be similar to what was done for the Production Workflow here: https://github.com/eclipse-pass/pass-documentation/blob/development/infrastructure-documenation/operations-production/ops-deploy.md

**Comment by tsande16** [comment]
> > May want to think about incorporating these documents into CI/CD documentation:
> > https://github.com/eclipse-pass/main/blob/main/docs/dev/release-steps-with-automations.md https://github.com/eclipse-pass/main/blob/main/docs/dev/release.md
> 
> I think this initial document is currently written as an overview section / landing page, that will transition the reader into reading more about the release process. We can structure the documentation to lead into the release and automation documents that you are referencing.

Decision during our PASS Weekly meeting was to give these documents their own section. [New issue created to address this](https://github.com/eclipse-pass/main/issues/1047).

**Comment by cshannon2** [comment]
Based on offline conversations we should add more detail regarding what happens when a PR is submitted. Then it should be ready to merge.

**Review by markpatton — APPROVED** [review]
I think this strikes a good balance between an overview and going deep into the weeds on the github action setup.

**Comment by cshannon2** [comment]
I have added additional information including: the PASS projects use of GitHub Actions, GitHub Action Triggers that are utilized, Details regarding what happens when a PR is submitted, and Related Documentation.

File: infrastructure-documenation/CI-CD.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Continuous deployment takes continuous delivery a step further. Continuous Deployment takes the deployment-ready build artifact that passed through the standardized test process and automatically deploys the artifact into a production environment. 
+
+### Continuous Deployment vs Continuous Delivery
+The difference between continuous delivery and continuous deployment is the presence of a manual approval to update to production. With continuous deployment, production happens automatically without explicit approval. 
+
+### CI/CD vs. Automation
+CI/CD is automating the build, release, and deployment process 
+All CI/CD utilizes automation but not all automation is CI/CD 
+
+## Current Status:
```
Comment (tsande16) [inline infrastructure-documenation/CI-CD.md:21]: Remove colon in headings for `Current Status`, `GitHub Actions`, and `Pull Request Submission Trigger`

**Review by tsande16 — COMMENTED** [review]
Overall this reads well. Small grammar quibbles. I think this should follow the pattern of the other sections and be put in it's own directory.

File: infrastructure-documenation/CI-CD.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### Pull Request Submission Trigger: 
+When a pull request is opened multiple GitHub Actions are triggered automatically. One of the GitHub Actions focuses on continuous integration concepts. The continuous integration GitHub Action runs several forms of tests. The following tests are run: Acceptance Tests, Unit Tests, and Integration Tests. Another GitHub Action is triggered during the opening of a pull request, the Git ECA Validation Status actioon. The Git ECA Validation Status action verifies that the author of the pull request is covered by necessary legal agreements to contribute to an Eclipse Foundation Project. Once the actions have completed the GitHub user interface shows if the tests passed or failed. 
+
+### CI/CD Workflow Proposal
+The team is in the process of implementing the following proposal workflow:
+<figure><img src="../.gitbook/assets/CI_CD_Workflow_Proposal.png" alt=""><figcaption></figcaption></figure>
+
+## Related Documentation:
+Release Process \
```
[the remark this answers is not in the document]
Comment (cshannon2) [reply infrastructure-documenation/CI-CD.md]: done

File: infrastructure-documenation/CI-CD.md (hunk 1 above)
Comment (cshannon2) [reply infrastructure-documenation/CI-CD.md]: done

**Comment by tsande16** [comment]
> Overall this reads well. Small grammar quibbles. I think this should follow the pattern of the other sections and be put in it's own directory.

Changes look good. One more thing, add this to it's own directory. Maybe something called "CI/CD" or spelled out: "Continuous Integration/Continuous Delivery".

**Comment by cshannon2** [comment]
> > Overall this reads well. Small grammar quibbles. I think this should follow the pattern of the other sections and be put in it's own directory.
> 
> Changes look good. One more thing, add this to it's own directory. Maybe something called "CI/CD" or spelled out: "Continuous Integration/Continuous Delivery".

Done.

**Comment by tsande16** [comment]
> > > Overall this reads well. Small grammar quibbles. I think this should follow the pattern of the other sections and be put in it's own directory.
> > 
> > 
> > Changes look good. One more thing, add this to it's own directory. Maybe something called "CI/CD" or spelled out: "Continuous Integration/Continuous Delivery".
> 
> Done.

My apologies, I was thinking of the user friendly names for `summary.md` when I wrote suggestions. The actual directory name should probably follow the convention of the other directory names and be something like `ci-cd`.

**Comment by cshannon2** [comment]
> > > > Overall this reads well. Small grammar quibbles. I think this should follow the pattern of the other sections and be put in it's own directory.
> > > 
> > > 
> > > Changes look good. One more thing, add this to it's own directory. Maybe something called "CI/CD" or spelled out: "Continuous Integration/Continuous Delivery".
> > 
> > 
> > Done.
> 
> My apologies, I was thinking of the user friendly names for `summary.md` when I wrote suggestions. The actual directory name should probably follow the convention of the other directory names and be something like `ci-cd`.

Done.

**Review by tsande16 — APPROVED** [review]
Looks good!

**Pull request merged**